src/controllers/router/index.tsx

```diff
@@ -59,16 +75,35 @@ Router.requestResources = async ({
   location,
   history,
   timeout,
+  loaders,
   ...bootstrapProps
 }: RequestResourcesParams) => {
-  const { bootstrapStore, requestRouteResources } = getRouterStore().actions;
+  const { bootstrapStore, loadRoute } = getRouterStore().actions;
+
+  const loader = (() => {
+    if (loaders) {
+      return combine(...loaders);
+    }
+
+    // default 'loaders' fallback for the first relase
+    const resources = createResourcesLoader({
+      context: bootstrapProps.resourceContext,
+      resourceData: null,
+      timeout,
+    });
+
+    return combine(resources);
+  })();
 
   bootstrapStore({
     ...bootstrapProps,
     history: history || createMemoryHistory({ initialEntries: [location] }),
+    loader,
   });
 
-  await requestRouteResources({ timeout });
+  // await requestRouteResources({ timeout });
+
+  await loadRoute().resources;
```

Comment:
What if I (as a consumer) have another loader that needs to be awaited?
Isn't it better if this is:
```js
await Promise.all(loadRoute())
```
And `loadRoute` is just a map of the calls of all loader `.load()` calls?

Reply:
Seems this is handled internally now